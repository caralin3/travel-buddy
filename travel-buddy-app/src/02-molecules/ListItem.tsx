import React from 'react';
import { ListGroupItem, ListGroupItemHeading, ListGroupItemText } from 'reactstrap';

export interface ListItemProps {
  dates?: string;
  description?: string;
  heading: string;
  onClick?: () => void;
}

export const ListItem: React.FC<ListItemProps> = ({ dates, description, heading, onClick }) => (
  <ListGroupItem className="list-item py-3" onClick={!!onClick ? onClick : undefined}>
    <ListGroupItemHeading className="list-item__heading text-primary">{heading}</ListGroupItemHeading>
    <ListGroupItemText className="list-item__body d-flex flex-column align-items-start m-0">
      {!!dates && <span className="list-item__dates">{dates}</span>}
      {!!description && (
        <span className="list-item__description mt-3" dangerouslySetInnerHTML={{ __html: description }} />
      )}
    </ListGroupItemText>
  </ListGroupItem>
);
