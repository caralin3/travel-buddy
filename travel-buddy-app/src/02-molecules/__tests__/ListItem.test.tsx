import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import { ListItem, ListItemProps } from '../ListItem';

const props: ListItemProps = {
  dates: '2022-03-24',
  description: 'Description',
  heading: 'Heading',
  onClick: jest.fn(),
};

describe('ListItem', () => {
  it('should render correctly', () => {
    const wrapper = shallow(<ListItem {...props} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
