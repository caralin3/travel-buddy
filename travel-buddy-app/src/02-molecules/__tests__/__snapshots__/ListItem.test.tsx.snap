// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListItem should render correctly 1`] = `
<ListGroupItem
  className="list-item py-3"
  onClick={[MockFunction]}
  tag="li"
>
  <ListGroupItemHeading
    className="list-item__heading text-primary"
    tag="h5"
  >
    Heading
  </ListGroupItemHeading>
  <ListGroupItemText
    className="list-item__body d-flex flex-column align-items-start m-0"
    tag="p"
  >
    <span
      className="list-item__dates"
    >
      2022-03-24
    </span>
    <span
      className="list-item__description mt-3"
      dangerouslySetInnerHTML={
        Object {
          "__html": "Description",
        }
      }
    />
  </ListGroupItemText>
</ListGroupItem>
`;
